examples/02_intermediate/wikimedia.ipynb

```diff
@@ -194,39 +200,53 @@
     "                        # each stream multiple times an hour. The presence of these canary\n",
     "                        # events in a stream allow us to differentiate between a broken event\n",
     "                        # stream, and an empty one.\n",
-    "                        if change['meta']['domain'] == 'canary':\n",
+    "                        # We will also filter bot-generated events\n",
+    "                        if change['meta']['domain'] == 'canary' or re.search('bot', change['user'], re.IGNORECASE):\n",
     "                            continue\n",
-    "                        print(f\"{change['meta']['dt']}:: {change['user']} edited {change['title']}\")\n",
+    "                        timestamp = change['meta']['dt']\n",
+    "                        event = f\"{timestamp}:: {change['user']} edited {change['title']}\"\n",
     "                        servername = change['server_name']\n",
     "                        if servername in servernames:\n",
     "                            servernames[servername] += 1\n",
     "                        else:\n",
     "                            servernames[servername] = 1\n",
-    "                        # Update perspective widget with this event\n",
-    "                        self.widget.update(\n",
-    "                            {\n",
-    "                                \"servername\": [change['server_name']],\n",
-    "                                \"servername_count\": [servernames[servername]],\n",
-    "                                \"time\": [change['meta']['dt']],\n",
-    "                                \"event\": [f\"{change['user']} edited {change['title']}\"],\n",
-    "                            }\n",
-    "                        )\n",
     "                        # Manually \"tick\" this edge\n",
-    "                        self.push_tick(event)\n",
+    "                        self.push_tick(\n",
+    "                            WikiData(\n",
+    "                                servername=servername,\n",
+    "                                servername_count=servernames[servername],\n",
+    "                                timestamp=timestamp,\n",
+    "                                event=event\n",
+    "                            )\n",
+    "                        )\n",
+    "\n",
+    "# Create the graph-time representation of our adapter\n",
+    "FetchWikiData = py_push_adapter_def(\"FetchWikiData\", FetchWikiDataAdapter, csp.ts[WikiData])\n",
+    "\n",
+    "@csp.node\n",
+    "def update_widget(wiki_event: csp.ts[WikiData], widget: PerspectiveWidget) -> csp.ts[str]:\n",
```

Comment:
I wouldnt have this return wiki_event.event, a node should really be responsible for one thing.
This is to update the widget, it should just update it and not output anything.
Also lets add an `if csp.ticked(wiki_event):` clause here since its better practice